aspnet/fundamentals/static-files.rst

```diff
@@ -1,8 +1,336 @@
+
+- wwwroot
+
+  - css
+  - images
+  - ...
+
+- MyStaticFiles
+
+  - test.png
+  - default.html
+
+Using the hierarchy example above, you might want to enable static files, default files, and browsing for the ``MyStaticFiles`` directory. In the following code snippet, that is accomplished with a single call to ``UseFileServer``.
+
+.. code-block:: c#
+
+  // Enable all static file middleware (serving of static files, default files,
+  // and directory browsing) for the MyStaticFiles directory.
+  app.UseFileServer(new FileServerOptions()
+  {
+      FileProvider = new PhysicalFileProvider(@"D:\Source\WebApplication1\src\WebApplication1\MyStaticFiles"),
+      RequestPath = new PathString("/StaticFiles"),
+      EnableDirectoryBrowsing = true
+  });
+
+Using the example hierarchy and code snippet from above, here's what happens if the user browses to various URIs:
+
+  - ``http://<yourApp>/StaticFiles/test.png`` - The ``MyStaticFiles/test.png`` file will be served to and presented by the browser.
+  - ``http://<yourApp>/StaticFiles`` - Since a default file is present (``MyStaticFiles/default.html``), that file will be served. If that file didn't exist, the browser would present a list of files in the ``MyStaticFiles`` directory (because the ``FileServerOptions.EnableDirectoryBrowsing`` property is set to ``true``).
+
+Working with content types
+--------------------------
+
+The ASP.NET static files middleware defines `almost 400 known file content types <https://github.com/aspnet/StaticFiles/blob/dev/src/Microsoft.AspNet.StaticFiles/FileExtensionContentTypeProvider.cs>`_. If the user attempts to reach a file of an unknown file type, ASP.NET will return a 404 Not Found error.
+
+Let's take the following directory/file hierarchy example to illustrate:
+
+- wwwroot
+
+  - css
+  - images
+
+    - test.image
+
+  - ...
+
+Using this hierarchy, you could enable static file serving and directory browsing with the following:
+
+.. code-block:: c#
+  :emphasize-lines: 5-6
+
+  public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
+  {
+    ...
+    // Serve static files and allow directory browsing.
+    app.UseDirectoryBrowser();
+    app.UseStaticFiles();
+
+If the user browses to ``http://<yourApp>/images``, a directory listing will be displayed by the browser that includes the ``test.image`` file. However, if the user clicks on that file, they will see a 404 error - even though the file obviously exists. In order to allow the serving of unknown file types, you could set the ``StaticFileOptions.ServeUnknownFileTypes`` property to ``true``:
+
+.. code-block:: c#
+  :emphasize-lines: 5-6
+
+  public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
+  {
+    ...
+    // Serve static files, including unknown file types, and allow directory browsing.
+    app.UseDirectoryBrowser();
+    app.UseStaticFiles(new StaticFileOptions { ServeUnknownFileTypes = true });
```

Comment:
I know you do that in the next example, but its a bad practice to send any response without a content-type header. Combine the samples.